lib/tables.c

```diff
@@ -3756,11 +4451,20 @@ table_collection_copy(table_collection_t *self, table_collection_t *dest)
     if (ret != 0) {
         goto out;
     }
+    ret = individual_table_copy(&self->individuals, &dest->individuals);
+    if (ret != 0) {
+        goto out;
+    }
+    ret = population_table_copy(&self->populations, &dest->populations);
+    if (ret != 0) {
+        goto out;
+    }
     ret = provenance_table_copy(&self->provenances, &dest->provenances);
     if (ret != 0) {
         goto out;
     }
     dest->sequence_length = self->sequence_length;
+    memcpy(dest->uuid, self->uuid, UUID4_LEN);
```

Comment:
copying should not change the uuid (since it's a property of a *file* not a table collection, really)